library/Fission/IPFS/Peer.hs

```diff
@@ -54,5 +62,45 @@ connect peer@(Peer peerID) = IPFSProc.run ["swarm", "connect"] (UTF8.textToLazyB
   (ExitFailure _ , _, _) -> Left $ CannotConnect peer
   (ExitSuccess   , _, _) -> Right ()
 
+peerAddressRe :: Regex
+peerAddressRe = mkRegex "^/ip[46]/([a-zA-Z0-9.:]*)/"
+
+extractIPfromPeerAddress :: String -> Maybe String
+extractIPfromPeerAddress peer = peer
+  & matchRegex peerAddressRe
+  <&> Data.List.head
+
+isExternalIPv4 :: Text -> Bool
+isExternalIPv4 ip = ip
+  & T.unpack
+  & extractIPfromPeerAddress
+  & maybe "" T.pack
+  & Net.IPv4.decode
+  <&> Net.IPv4.reserved
+  & maybe False not
+
+
+filterExternalPeers :: [Peer] -> [Peer]
+filterExternalPeers = Data.List.filter (isExternalIPv4 . peer)
+
+getExternalAddress :: (MonadIO m
+  , MonadReader cfg m
+  , HasProcessContext cfg
+  , HasLogFunc cfg
+  , Has IPFS.BinPath cfg
+  , Has IPFS.Timeout cfg)
+  => m (Either Error [Peer])
+getExternalAddress = IPFSProc.run' ["id"] >>= \case
+    (ExitFailure _ , _, err) ->
+      return $ Left $ UnknownErr $ "BOOM:" <> UTF8.textShow err
+
+    (ExitSuccess , rawOut, _) -> do
+      let ipfsInfo = JSON.decode rawOut
+      let filteredAddrs = filterExternalPeers . maybe [] _addresses $ ipfsInfo
```

Comment:
💅 

`let` on each line is completely fine, but it tends to be broken into a grouped assignment section:

```haskell
(ExitSuccess , rawOut, _) -> 
  let
    ipfsInfo = JSON.decode rawOut
    filteredAddrs = filterExternalPeers . maybe [] _addresses $ ipfsInfo

  logInfo $ displayShow ipfsInfo
  return $ Right filteredAddrs
```

It's basically saying "here's my variables ... and now here's how they're used"

Reply:
Ahh yeah I like this much better ❤️